src/choose-location/choose-location.resource.ts

```diff
@@ -0,0 +1,18 @@
+import { openmrsFetch, openmrsObservableFetch } from "@openmrs/esm-api";
+import { map } from "rxjs/operators";
+
+export function getLoginLocations(): Observable<object> {
+  return openmrsObservableFetch(
+    `/ws/rest/v1/location?tag=Login%20Location&v=custom:(uuid,display)`
+  ).pipe(map(({ data }) => data["results"]));
+}
+
+export function setSessionLocation(locationUuid: string): Promise<any> {
```

Comment:
It's good practice to make network requests cancelable. Cleaning up network requests in effects avoids memory leaks. To do so, you have two options:

Option 1
```suggestion
export function setSessionLocation(locationUuid: string, abortController: AbortController): Promise<any> {
  return openmrsFetch("/ws/rest/v1/appui/session", { signal: abortController.signal })
```

Option 2: use `openmrsObservableFetch` (https://wiki.openmrs.org/display/projects/openmrs-esm-api). This automatically aborts the network request when you unsubscribe.